src/canisters/frontend/ic-certified-assets/src/state_machine.rs

```diff
@@ -36,12 +44,137 @@ pub struct AssetEncoding {
     pub sha256: [u8; 32],
 }
 
+#[derive(Deserialize, CandidType, Serialize, Debug, Default, Clone, PartialEq, Eq)]
+pub struct RedirectUrl {
+    pub(crate) host: Option<String>,
+    pub(crate) path: Option<String>,
+}
+
+#[derive(Default, Clone, Debug, CandidType, Deserialize)]
+pub struct AssetRedirect {
+    pub from: Option<RedirectUrl>,
+    pub to: RedirectUrl,
+    pub user_agent: Option<Vec<String>>,
+    pub response_code: u16,
+}
```

Comment:
I have had a hard time coming up with the right place to put these types (the choice between `state_machine.rs` and `types.rs`